Packages/com.unity.inputsystem/InputSystem/Plugins/XInput/XboxOneGamepad.cs

```diff
@@ -161,6 +208,76 @@ protected override void FinishSetup(InputDeviceBuilder builder)
             paddle2 = builder.GetControl<ButtonControl>(this, "paddle2");
             paddle3 = builder.GetControl<ButtonControl>(this, "paddle3");
             paddle4 = builder.GetControl<ButtonControl>(this, "paddle4");
+        }
+
+        public ulong gamepadId
+        {
+            get
+            {
+                UpdatePadSettings();
+                return m_GamepadId;
+            }
+        }
+
+        public int xboxOneUserId
+        {
+            get
+            {
+                UpdatePadSettings();
+                return m_XboxOneUserId;
```

Comment:
This should be part of the device configuration. There's no way with the current API to know when the user has changed other than to constantly poll the value. Configuration getters should retrieve the current configuration via a device command / IOCTL and then cache it. When the configuration changes, the backend should send a configuration change event.